src/follower.py

```diff
@@ -106,11 +113,71 @@ def stop(self):
+            # Left condition: when the turn angle is between 90 + deadzone and 135, turn the wheels at that angle plus the offset. 
+            # If the turn angle is over the maximum wheel turn angle, set the turn angle to the wheel max.
+            self._fw.turn(self._turn_angle + self.camera_angle_offset) 
+            if self._turn_angle + self.camera_angle_offset > self.WHEEL_MAX:
+              self._fw.turn(self.WHEEL_MAX)
+
+        elif self.STRAIGHT_ANGLE - self.DEADZONE_WHEELS > self._turn_angle:
+            # Right condition: when the turn angle is between 90 - deadzone and 45, turn the wheels at that angle minus the offset. 
+            # If the turn angle is under the minimum wheel turn angle, set the turn angle to the wheel min.
+            self._fw.turn(self._turn_angle - self.camera_angle_offset)
+            if self._turn_angle - self.camera_angle_offset < self.WHEEL_MIN:
+              self._fw.turn(self.WHEEL_MIN)
+
+
+    def pan_camera(self):
+        """
+        Follows the tag by panning camera towards the same direction as the wheels.
+        """
+        if self._turn_angle < self.STRAIGHT_ANGLE - self.DEADZONE_CAMERA:
+            # Left condition: If the turn angle is less than 90 - deadzone, turn the camera left. 
+            # The offset is 90 - the current angle.
+            self.turn_camera_left(self._turn_angle)
+            self.camera_angle_offset = np.abs(90 - self._camera.current_pan)
+
+        elif self._turn_angle > self.STRAIGHT_ANGLE + self.DEADZONE_CAMERA:
+            # Right condition: If the turn angle is greater than 90 + deadzone, turn the camera right. 
+            # The offset is the current angle - 90.
+            self.turn_camera_right(self._turn_angle)
+            self.camera_angle_offset = np.abs(self._camera.current_pan - 90)
+
+        if self.camera_angle_offset == self.STRAIGHT_ANGLE or self.camera_angle_offset == self.WHEEL_MIN:
+            # If the offset of the right or left offsets are equal to 90 or 45 respectively, turn the camera straight since the wheels will be straight.
+            self._camera.current_pan = self.STRAIGHT_ANGLE
+            self._camera.pan_servo.write(self._camera.current_pan)
+            self._fw.turn(self._camera.current_pan)
+ 
+
+    def turn_camera_left(self, angle):
+        """
+        Turns camera left. Camera API turn left function only takes steps.
+        """
+        self._camera.turn_left(self.angle_to_step())
+    
+    def turn_camera_right(self, angle):
+        """
+        Turns camera right. Camera API turn right function only takes steps.
+        """
+        self._camera.turn_right(self.angle_to_step())
 
+    
+    def angle_to_step(self):
+        """
+        Definition for converting angles to steps. 
+        Takes the angle property and converts into steps where 1 step is 5 degrees.
+        """
+        return (self._turn_angle/self._camera.PAN_STEP) + 4
```

Comment:
What is the purpose of adding a 4 here? Doesn't seem right.

Reply:
I added 4 (magic number) to get the camera to turn a bit more than what it was already doing.